src/transformers/modeling_utils.py

```diff
@@ -2344,15 +2345,37 @@ def from_pretrained(
         else:
             model_kwargs = kwargs
 
+        has_bnb_quantization_config = False
+        has_gptq_quantization_config = False
+        if hasattr(config, "quantization_config"):
+            # backward compatibility
+            has_bnb_quantization_config = (
+                hasattr(config.quantization_config, "load_in_8bit")
+                or config.quantization_config.get("quant_method", None) == "bitsandbytes"
+            )
+            has_gptq_quantization_config = config.quantization_config.get("quant_method", None) == "gptq"
+
+        gtpq_quantizer = None
+        if has_gptq_quantization_config:
+            if not (is_optimum_available() and is_auto_gptq_available()):
+                raise ImportError(
+                    "Loading GTPQ quantized model requires optimum library : `pip install optimum` and auto-gptq library 'pip install auto-gptq'"
+                )
+            else:
+                # Need to protect the import
+                from optimum.gptq import GPTQQuantizer
+            gtpq_quantizer = GPTQQuantizer.from_dict(config.quantization_config)
+            torch_dtype = config.torch_dtype
+
         if is_8bit_serializable and quantization_config is not None and load_in_8bit:
-            if hasattr(config, "quantization_config"):
+            if has_bnb_quantization_config:
```

Comment:
So here we would test `quantization_method=="bitandbytes"`, or better, an element of an enum `QuantizationMethod` to avoid typos (like `QuantizationMethod.BITS_AND_BYTES`).